src/evidently/collector/config.py

```diff
@@ -53,6 +53,22 @@ def is_ready(self, config: "CollectorConfig", storage: "CollectorStorage") -> bo
             self.last_triggered = now
             return True
         return False
+    
+
+class RowsCountTrigger(CollectorTrigger):
+    rows_count: int = 1
+
+    def is_ready(self, config: "CollectorConfig", storage: "CollectorStorage") -> bool:
+        print("---")
+        print("RowsCountTrigger")
+        print(f"rows_count: {self.rows_count}")
+        print(f"buffer length:{len(storage._buffers[config.id])}")
+        if len(storage._buffers[config.id]) > 0: 
```

Comment:
`storage` might not have `_buffers` field, only `InMemoryStorage` has it, which is only an implementation (there are no other implementations, but still).
You can introduce a new abstract method to `CollectorStorage` like `def get_buffer_size(self, id)` and use it here instead (and implement for in-memory storage with `len(self._buffers[id])`)

Reply:
True, it's safer to implement the buffer size as abstract method to force any future storage implementation to provide it. Thanks for the hint :) Done in 78819402f55eb983751b641ac393c1478bfe8d03 and retested on my local machine.